test/test_data_utils.py

```diff
@@ -109,3 +111,74 @@ def test_generate_sample_fastq_gzip_content_with_gunzip():
             binary_content = fp.read()
         unzipped = gunzip_content(content=binary_content)
         assert unzipped == SAMPLE_CONTENT_2x10
+
+
+def test_normalize_suffixes():
+
+    assert normalize_suffixes("foo", FASTQ_SUFFIXES) == ("foo.fastq", False)
+    assert normalize_suffixes("foo", FASTQ_SUFFIXES, compressed=None) == ("foo.fastq", False)
+    assert normalize_suffixes("foo", FASTQ_SUFFIXES, compressed=False) == ("foo.fastq", False)
+    assert normalize_suffixes("foo", FASTQ_SUFFIXES, compressed=True) == ("foo.fastq.gz", True)
+
+    assert normalize_suffixes("foo.gz", FASTQ_SUFFIXES) == ("foo.fastq.gz", True)
+    assert normalize_suffixes("foo.gz", FASTQ_SUFFIXES, compressed=None) == ("foo.fastq.gz", True)
+    assert normalize_suffixes("foo.gz", FASTQ_SUFFIXES, compressed=True) == ("foo.fastq.gz", True)
+
+    with pytest.raises(RuntimeError):  # This case is self-contradictory so raises an error.
+        normalize_suffixes("foo.gz", FASTQ_SUFFIXES, compressed=False)
+
+    assert normalize_suffixes("foo.fastq", FASTQ_SUFFIXES) == ("foo.fastq", False)
+    assert normalize_suffixes("foo.fastq", FASTQ_SUFFIXES, compressed=None) == ("foo.fastq", False)
+    assert normalize_suffixes("foo.fastq", FASTQ_SUFFIXES, compressed=False) == ("foo.fastq", False)
+    assert normalize_suffixes("foo.fastq", FASTQ_SUFFIXES, compressed=True) == ("foo.fastq.gz", True)
+
+
+def test_c4_278():
```

Comment:
Can we use a better descriptor (or put some info in the docstring)? I think referencing the ticket makes sense though.

Reply:
Well, I think if we had a convention it might be easier. If the name was `test_jira_c4_278`, would that be enough?